### PR TITLE
Add shared display widgets and interaction primitives

### PR DESCRIPTION
**Pull request by janosh**
Extract reusable display and interaction code into the widgets package so downstream apps can share implementations without adding runtime dependencies.

- Add `CodeBlock` with cancellable highlighting, escaped tokens or trusted HTML, stale-result protection, wrapping and visible errors.
- Add `StatGrid` with custom value formatting, units, visible hints and accessible deltas whose positive/negative meaning is explicitly configured.
- Add `Spinner`, dismissible `StatusMessage`, `DragOverlay` and repeatable `ClickFeedback`, including loading announcements and reduced-motion styles.
- Add `/canvas` for parent content-box sizing, DPR tracking, coalesced base/overlay frames, isolated drawing contexts and lifecycle cleanup that restores inline dimensions.
- Add `/roving-focus` for keyed HTML/SVG items, nested groups, DOM-order navigation and reactive disabled/hidden eligibility. Share availability checks with focus traps and fix focusable closed `<details>` elements being excluded.
- Add `/csv` with escaped records and optional explicit columns for sparse or header-only exports; add `/format` and reuse its byte formatter in the JSON tree.
- Add `/url-params` with typed query validation, boolean/default encoding and URL updates that preserve unrelated parameters and skip unchanged writes.
- Add `/image-markup` for contained-image geometry, palettes and canvas/OffscreenCanvas stroke rendering, plus `/stats` formatting helpers.
- Document the APIs, add workbench demos, exercise public package imports/types and cover cancellation, cleanup, accessibility and edge cases with focused regression tests.

### CODE REVIEW

File: src/lib/csv.ts [hunk 1]
```diff
@@ -0,0 +1,25 @@
+export type CsvCell = number | string | boolean | null | undefined
+
+// Quote a CSV field per RFC 4180. A lone CR counts: readers treat CR, LF and CRLF alike
+// as record separators, so an unquoted CR splits one record into two.
+export const escape_csv_field = (value: CsvCell): string => {
+  const field = String(value ?? ``)
```
[the remark this answers is not in the document]
Comment (janosh) [reply src/lib/csv.ts:8]: Keeping these helpers as data-preserving CSV serializers: prefixing by default would change negative numbers and literal strings consumed by downstream data pipelines. c7c1320 documents that RFC quoting does not prevent formula execution and adds regression cases preserving all four prefixes and negative numeric cells. Spreadsheet-specific sanitization belongs at the export call site, where the intended reader is known. OWASP describes the data-altering tradeoff and notes that no strategy is universal across spreadsheet applications and other consumers: https://owasp.org/www-community/attacks/CSV_Injection

File: tests/vitest/csv.test.ts [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+  })
+
+  test.each([
+    [`plain`, `plain`],
+    [42, `42`],
+    [-42, `-42`],
+    [`=1+2`, `=1+2`],
+    [`+123`, `+123`],
+    [`-123`, `-123`],
+    [`@name`, `@name`],
```
[the remark this answers is not in the document]
Comment (janosh) [reply tests/vitest/csv.test.ts:41]: This duplicates the CSV serialization discussion, whose data-preserving contract was subsequently accepted here: https://github.com/janosh/svelte-widgets/pull/455#discussion_r3936009843. These tests intentionally preserve literal string prefixes as well as negative numbers. Spreadsheet-specific sanitization remains the responsibility of the export call site, as documented in escape_csv_field.
